### PR TITLE
Openstack Swift integration

### PR DESCRIPTION
**Pull request by mynameisguy**
Addition of support for persisting Kafka logs to OpenStack Swift.
This includes support for keystone and tempAuth (and for the Softlayer object service)

### CODE REVIEW

**Comment by mynameisguy** [comment]
This addition was tested successfully using Amazon S3 service and Openstack swift.
For some reason, the tests fail when using fakeS3.

File: src/main/java/com/pinterest/secor/uploader/Uploader.java [hunk 1]
```diff
@@ -67,21 +67,40 @@ public Uploader(SecorConfig config, OffsetTracker offsetTracker, FileRegistry fi
     }
 
     private Future<?> upload(LogFilePath localPath) throws Exception {
-        String s3Prefix = "s3n://" + mConfig.getS3Bucket() + "/" + mConfig.getS3Path();
-        LogFilePath s3Path = new LogFilePath(s3Prefix, localPath.getTopic(),
+    	String prefix = "";
+        if (mConfig.getCloudService() != null && mConfig.getCloudService().equals("Swift")){ // the first condition
```
Comment (pgarbacki) [inline src/main/java/com/pinterest/secor/uploader/Uploader.java]: Could you please fix indentation across your changes?

File: src/main/config/secor.dev.properties [hunk 2]
```diff
@@ -4,9 +4,25 @@ include=secor.common.properties
...
+
 # Name of the s3 bucket where log files are stored.
 secor.s3.bucket=
 
+###############
+# Using Swift #
+###############
+
+# Boolean variable which determines if each topic will be uploaded to different container
+# (Created automatic) - if true the next setting will be ignored
```
Comment (pgarbacki) [inline src/main/config/secor.dev.properties]: What does "(Created automatic)" mean?

Comment (pgarbacki) [inline src/main/config/secor.dev.properties]: Clarify what "next setting" means.

File: src/main/config/secor.common.properties [hunk 3]
```diff
@@ -20,10 +20,30 @@
 # Regular expression matching names of consumed topics.
 secor.kafka.topic_filter=.*
 
+# Choose what to fill according to the service you are using
+# in the choice option you MUST fill S3 or Swift
+
+#cloud.service=Swift
```
Comment (pgarbacki) [inline src/main/config/secor.common.properties]: Remove this line.

File: src/main/java/com/pinterest/secor/parser/PartitionFinalizer.java [hunk 4]
```diff
@@ -85,8 +85,19 @@ public PartitionFinalizer(SecorConfig config) throws Exception {
         return mMessageParser.getFinalizedUptoPartitions(lastMessages, committedMessages);
     }
 
-    private void finalizePartitionsUpTo(String topic, String[] uptoPartitions) throws Exception {
-        final String s3Prefix = "s3n://" + mConfig.getS3Bucket() + "/" + mConfig.getS3Path();
+    private void finalizePartitionsUpTo(String topic, String[] uptoPartitions) throws Exception { 
+        String prefix = "";
+        if (mConfig.getCloudService() != null && mConfig.getCloudService().equals("Swift")) {
```
Comment (pgarbacki) [inline src/main/java/com/pinterest/secor/parser/PartitionFinalizer.java]: I thought that cloud.service must be set. In that case it's OK to fail with NullPointerException. Remove mConfig.getCloudService() != null.

File: src/main/java/com/pinterest/secor/util/FileUtil.java [hunk 5]
```diff
@@ -109,6 +123,16 @@ public static void touch(String path) throws IOException {
         fs.create(fsPath).close();
     }
 
+    public static void CreateContainer(String topic) throws IOException {
```
Comment (pgarbacki) [inline src/main/java/com/pinterest/secor/util/FileUtil.java]: createContainer

Comment (pgarbacki) [inline src/main/java/com/pinterest/secor/util/FileUtil.java]: I don't think we really need this method. Simply inline it.

File: src/main/java/com/pinterest/secor/uploader/Uploader.java (hunk 1 above)
Comment (pgarbacki) [inline src/main/java/com/pinterest/secor/uploader/Uploader.java]: Could you move this code to a separate method called getPrefix or something similar?

File: src/main/java/com/pinterest/secor/uploader/Uploader.java [hunk 6]
```diff
@@ -67,21 +67,40 @@ public Uploader(SecorConfig config, OffsetTracker offsetTracker, FileRegistry fi
     }
 
     private Future<?> upload(LogFilePath localPath) throws Exception {
```
Comment (pgarbacki) [inline src/main/java/com/pinterest/secor/uploader/Uploader.java]: Could you add some unit tests for this new logic?

**Comment by pgarbacki** [comment]
@mynameisguy you will need to fix the integration tests.

**Comment by pgarbacki** [comment]
@mynameisguy This looks good. Would you mind squashing your commits before I merge it?

**Comment by mynameisguy** [comment]
@pgarbacki Done

**Comment by pgarbacki** [comment]
Thank you for the contribution @mynameisguy

**Comment by pgarbacki** [comment]
BTW. @mynameisguy feel free to add yourself to the list of contributors.

**Pull request merged**